src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToArrayTransformer.php

```diff
@@ -0,0 +1,173 @@
+        }
+        $this->fields = $fields;
+        $this->pad = (bool) $pad;
+    }
+
+    /**
+     * Transforms a normalized date interval into an interval array.
+     *
+     * @param \DateInterval $dateInterval Normalized date interval.
+     *
+     * @return array Interval array.
+     *
+     * @throws UnexpectedTypeException If the given value is not a \DateInterval instance.
+     */
+    public function transform($dateInterval)
+    {
+        if (null === $dateInterval) {
+            return array_intersect_key(
+                array(
+                    'years' => '',
+                    'months' => '',
+                    'weeks' => '',
+                    'days' => '',
+                    'hours' => '',
+                    'minutes' => '',
+                    'seconds' => '',
+                    'invert' => false,
+                ),
+                array_flip($this->fields)
+            );
+        }
+        if (!$dateInterval instanceof \DateInterval) {
+            throw new UnexpectedTypeException($dateInterval, '\DateInterval');
+        }
+        $result = array();
+        foreach (self::$availableFields as $field => $char) {
+            $result[$field] = $dateInterval->format('%'.($this->pad ? strtoupper($char) : $char));
+        }
+        if (in_array('weeks', $this->fields, true)) {
+            $result['weeks'] = 0;
+            if (isset($result['days']) && (int) $result['days'] >= 7) {
+                $result['weeks'] = (string) floor($result['days'] / 7);
+                $result['days'] = (string) ($result['days'] % 7);
+            }
+        }
+        $result['invert'] = '-' === $result['invert'];
+        $result = array_intersect_key($result, array_flip($this->fields));
+
+        return $result;
+    }
+
+    /**
+     * Transforms an interval array into a normalized date interval.
+     *
+     * @param array $value Interval array
+     *
+     * @return \DateInterval Normalized date interval
+     *
+     * @throws UnexpectedTypeException       If the given value is not an array.
+     * @throws TransformationFailedException If the value could not be transformed.
+     */
+    public function reverseTransform($value)
+    {
+        if (null === $value) {
+            return;
+        }
+        if (!is_array($value)) {
+            throw new UnexpectedTypeException($value, 'array');
+        }
+        if ('' === implode('', $value)) {
+            return;
+        }
+        $emptyFields = array();
+        foreach ($this->fields as $field) {
+            if (!isset($value[$field])) {
+                $emptyFields[] = $field;
+            }
+        }
+        if (count($emptyFields) > 0) {
+            throw new TransformationFailedException(sprintf('The fields "%s" should not be empty', implode('", "', $emptyFields)));
+        }
+        if (isset($value['invert']) && !is_bool($value['invert'])) {
+            throw new UnexpectedTypeException($value['invert'], 'boolean');
```

Comment:
The error message is going to be `Expected argument of type "boolean", xxx" given`. I don't think there will be anything in the context mentioning that the error comes from the `invert` value and not the whole value for the form.